core/drivers/rockchip_otp.c

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) 2025, Pengutronix, Michael Tretter <m.tretter@pengutronix.de>
+ */
+
+#include <common.h>
+#include <io.h>
+#include <kernel/panic.h>
+#include <kernel/tee_common_otp.h>
+#include <mm/core_memprot.h>
+#include <utee_defines.h>
+
+#include <drivers/rockchip_otp.h>
+
+#define OTP_S_AUTO_CTRL	0x0004
+#define OTP_S_AUTO_EN	0x0008
+#define OTP_S_PROG_DATA	0x0010
+#define OTP_S_DOUT	0x0020
+#define OTP_S_INT_ST	0x0084
+
+#define ADDR_SHIFT	16
+#define BURST_SHIFT	8
+#define CMD_READ	0
+#define CMD_WRITE	2
+#define EN_ENABLE	1
+#define EN_DISABLE	0
+
+#define MAX_INDEX	0x300
+#define BURST_SIZE	8
+#define OTP_WORD	1
+
+#define OTP_S_ERROR_BIT		BIT32(4)
+#define OTP_S_WR_DONE_BIT	BIT32(3)
+#define OTP_S_VERIFY_BIT	BIT32(2)
+#define OTP_S_RD_DONE_BIT	BIT32(1)
+
+#define OTP_POLL_PERIOD_US	0
+#define OTP_POLL_TIMEOUT_US	1000
+
+TEE_Result tee_otp_read_secure(uint32_t *value, uint32_t index,
+			       uint32_t count)
```

Comment:
I kept the function name that was used in the rk3588 platform implementation. If this is a separate driver, the function name probably should be something like `rockchip_otp_read`. Thoughts?

Reply:
I agree, it would be much better. Could you rename the functions?